hamilton/function_modifiers/base.py

```diff
@@ -99,18 +101,40 @@ def __call__(self, fn: Callable):
         :param fn: Function to decorate
         :return: The function again, with the desired properties.
         """
-        self.validate(fn)
+        # # stop unwrapping if not a hamilton function
+        # # should only be one level of "hamilton wrapping" - and that's what we attach things to.
+        self.validate(unwrap(fn, stop=lambda f: not hasattr(f, "__hamilton__")))
```

Comment:
Why do we only do one level deep? Aren't there some weird cases with multiple imports, imports of imports, etc...?

Reply:
Yes, I got that.  Why *just* one layer was the question?

Reply:
> Why just one layer 

1. Doesn't seem like we need more? 
2. This mirrors the prior behavior of directly attaching to the function - there were no layers before.